script.js

```diff
@@ -107,68 +110,75 @@ const oper = {
     },
 };
 
-
-function shuntingYard(expr) {
+//converts to postfix expression
+function convertExpression(expr) {
     const output = [];
     const stack = [];
     let accum = '';
+
     for (let i = 0; i < expr.length; i++) {
+        let top = stack.length - 1;
+        const char = expr[i];
+        const next = expr[i + 1];
 
-        if (isNaN(expr[i]) && expr[i] !== '.') {
-            if (expr[i] === '(' || !stack.length || stack[stack.length - 1] === '(') {
-                stack.push(expr[i]);
-            } else if (expr[i] !== ')') {
-                while (stack[stack.length - 1] !== '(' && stack.length && oper[expr[i]].precedence <= oper[stack[stack.length - 1]].precedence) {
+        if (isNaN(char) && char !== '.') {
+
+            if (!stack.length || char === '(' || stack[top] === '(') {
+                stack.push(char);
+            } else if (oper.hasOwnProperty(char)) {
+                while (stack.length && stack[top] !== '(' && oper[char].precedence <= oper[stack[top]].precedence) {
                     output.push(stack.pop());
```

Comment:
much cleaner, remember scope though- each iteration the 'top' variable is an entirely new box, unless you are reassigning it later (cant see the later code on this diff) it can be const